### ISSUE TITLE
[SmartAI] Follow action does not include a catchup mechanic

### ISSUE THREAD

**Issue by Gultask** [azerothcore/azerothcore-wotlk#14698]
### Current Behaviour

When using the SAI's FOLLOW action to make a creature follow another, the follower will lag behind, since they'll not speed up to walk/run alongside the master creature.

### Expected Blizzlike Behaviour

The follower constantly has an update pulse go off, at exactly every 1.2 seconds (3 packet cycles), and follow the leader based on that, it creates the behavior shown below and all over the game, creating a staggered movement.

https://www.youtube.com/watch?v=I16uIzEBQe8&t=124s

Thanks to Ovahlord and Neifion for providing this info.

In short: **it should behave like formations**

### Steps to reproduce the problem

1. `.go c id 10676` one raider will lag behind
2. `.go c id 21244` follow the messenger, see that the Marauders hang behind
3. `.go c id 20353` See that Helcular lags behind

### Extra Notes

Related to: https://github.com/azerothcore/azerothcore-wotlk/issues/14274

### AC rev. hash/commit

https://github.com/azerothcore/azerothcore-wotlk/commit/b3e37226745a6e6de308bb7b00f0d2f9c596938e

### Operating system

Windows 10
